lib/web_ui/dev/test_runner.dart

```diff
@@ -186,6 +181,20 @@ class TestCommand extends Command<bool> with ArgUtils {
     return true;
   }
 
+  /// Preparations before running the tests such as booting simulators or
+  /// creating directories.
+  Future<void> _prepare() async {
+    if(!environment.webUiTestResultsDirectory.existsSync()) {
+      environment.webUiTestResultsDirectory.createSync(recursive: true);
+    }
```

Comment:
How about:

```dart
    if(environment.webUiTestResultsDirectory.existsSync()) {
      environment.webUiTestResultsDirectory.delete(recursive: true);
    }
```

Reply:
delete can be brittle, if OS still holding on to directory somehow.

Reply:
You mean we should first delete then create the directory? 

It does not make a difference in LUCI since we have hermetic tests (the entire engine directory is created minutes before the tests and all dependencies are added before the tests) actually start running, but I can add if you think it might be useful for local development.

Reply:
I'm not sure I understand this suggestion. We want to make sure the directory exists before running the tests. And you are suggesting we delete it?

Reply:
Or are you asking to delete the directory after the test is done?

Reply:
sorry I just saw this. The recipe code is here: https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/web_engine.py#245

we don't delete the directory, it is uploaded to gcs bucket.